components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PatientFamilyLockModule.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.studies.family.internal;
+
+import org.phenotips.data.Patient;
+import org.phenotips.data.PatientRepository;
+import org.phenotips.studies.family.Family;
+import org.phenotips.studies.family.FamilyRepository;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.context.Execution;
+import org.xwiki.locks.DocumentLock;
+import org.xwiki.locks.LockModule;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.users.User;
+import org.xwiki.users.UserManager;
+
+import java.util.Collections;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.doc.XWikiLock;
+
+/**
+ * Prevents editing of a patient when the patient's family doc is locked.
+ *
+ * @version $Id$
+ * @since 1.3M1
+ */
+@Component
+@Singleton
+@Named("patientfamlock")
+public class PatientFamilyLockModule implements LockModule
+{
+
+    @Inject
+    private Execution execution;
+
+    @Inject
+    private UserManager userManager;
+
+    @Inject
+    private FamilyRepository familyRepository;
+
+    @Inject
+    private PatientRepository patientRepository;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public int getPriority()
+    {
+        return 300;
```

Comment:
I agree with Itai that resolving this ambiguity manually is better for two reasons:
1) who knows how ambiguities are going to be resolve din the future version of XWiki
2) someone who debugs the code X years from now may not know this rule and waste some time figuring this out in case of some problem